tests/integration/local/start_api/test_start_api_with_terraform_application.py

```diff
@@ -0,0 +1,56 @@
+import shutil
+import os
+from pathlib import Path
+from typing import Optional
+from unittest import skipIf
+from http.client import HTTPConnection
+
+import pytest
+import requests
+
+from tests.integration.local.start_api.start_api_integ_base import StartApiIntegBaseClass
+from tests.testing_utils import get_sam_command, CI_OVERRIDE
+
+
+class TerraformStartApiIntegrationBase(StartApiIntegBaseClass):
+    terraform_application: Optional[str] = None
+
+    @classmethod
+    def setUpClass(cls):
+        command = get_sam_command()
+        cls.template_path = ""
+        cls.build_before_invoke = False
+        cls.command_list = [command, "local", "start-api", "--hook-name", "terraform", "--beta-features"]
+        cls.test_data_path = Path(cls.get_integ_dir()) / "testdata" / "start_api"
+        cls.project_directory = cls.test_data_path / "terraform" / cls.terraform_application
+        super(TerraformStartApiIntegrationBase, cls).setUpClass()
+
+    @staticmethod
+    def get_integ_dir():
+        return Path(__file__).resolve().parents[2]
+
+    def tearDown(self) -> None:
+        shutil.rmtree(str(Path(self.project_directory / ".aws-sam-iacs")), ignore_errors=True)  # type: ignore
+        shutil.rmtree(str(Path(self.project_directory / ".terraform")), ignore_errors=True)  # type: ignore
+        try:
+            os.remove(str(Path(self.project_directory / ".terraform.lock.hcl")))  # type: ignore
+        except (FileNotFoundError, PermissionError):
+            pass
+
+
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+@pytest.mark.flaky(reruns=3)
+class TestStartApiTerraformApplication(TerraformStartApiIntegrationBase):
```

Comment:
Did we usually set a pytest timeout for these tests? Or did we usually rely on the `start-api` command to time us out?

Reply:
We have a timeout configured in the `wait_for_local_process` function that starts up the process in the tests. Is that what you're referring to?

https://github.com/aws/aws-sam-cli/blob/develop/tests/integration/local/common_utils.py#L8

Reply:
More so with after the API loads and we make a request + wait for response

Reply:
Oh scratch what I said, the request has it's own timeout defined 😅